src/components/Modal/index.tsx

```diff
@@ -1,69 +1,199 @@
-
-import { sm } from 'src/theme/variables'
-
-const useStyles = makeStyles(
-  createStyles({
-    root: {
-      alignItems: 'center',
-      flexDirection: 'column',
-      display: 'flex',
-      overflowY: 'scroll',
-    },
-    paper: {
-      position: 'relative',
-      top: '68px',
-      width: '500px',
-      borderRadius: sm,
-      backgroundColor: '#ffffff',
-      boxShadow: '0 0 5px 0 rgba(74, 85, 121, 0.5)',
-      '&:focus': {
-        outline: 'none',
-      },
-      display: 'flex',
-      flexDirection: 'column',
-    },
-  }),
-)
+import React, { ReactElement, ReactNode, ReactNodeArray } from 'react'
+import { StyledScrollableBar } from 'src/routes/safe/components/Transactions/TxList/styled'
+import styled from 'styled-components'
+
+const ModalStyled = styled(ModalMUI)`
+  & {
+    align-items: center;
+    flex-direction: column;
+    display: flex;
+    overflow-y: scroll;
+  }
+
+  .paper {
+    position: relative;
+    top: 68px;
+    width: 500px;
+    border-radius: 8px;
+    background-color: #ffffff;
+    box-shadow: 0 0 5px 0 rgba(74, 85, 121, 0.5);
+    display: flex;
+    flex-direction: column;
+
+    &:focus {
+      outline: none;
+    }
+
+    // TODO: replace class-based styles by params
+    &.receive-modal {
+      height: auto;
+      max-width: calc(100% - 130px);
+      min-height: 544px;
+      overflow: hidden;
+    }
+
+    &.bigger-modal-window {
+      width: 775px;
+      height: auto;
+    }
+
+    &.smaller-modal-window {
+      height: auto;
+    }
+
+    &.modal {
+      height: auto;
+      max-width: calc(100% - 130px);
+      overflow: hidden;
+    }
+  }
+`
 
 interface GnoModalProps {
   children: ReactNode
   description: string
   // type copied from Material-UI Modal's `close` prop
-  handleClose?: {
-    bivarianceHack(event: Record<string, unknown>, reason: 'backdropClick' | 'escapeKeyDown'): void
-  }['bivarianceHack']
-  modalClassName?: string
+  handleClose?: (event: Record<string, unknown>, reason: 'backdropClick' | 'escapeKeyDown') => void
   open: boolean
   paperClassName?: string
   title: string
 }
 
-const GnoModal = ({
-  children,
-  description,
-  handleClose,
-  modalClassName,
-  open,
-  paperClassName,
-  title,
-}: GnoModalProps): ReactElement => {
-  const classes = useStyles()
-
+const GnoModal = ({ children, description, handleClose, open, paperClassName, title }: GnoModalProps): ReactElement => {
   return (
-    <Modal
-      aria-describedby={description}
-      aria-labelledby={title}
-      className={cn(classes.root, modalClassName)}
-      onClose={handleClose}
-      open={open}
-    >
-      <div className={cn(classes.paper, paperClassName, 'classpep')}>{children}</div>
-    </Modal>
+    <ModalStyled aria-describedby={description} aria-labelledby={title} onClose={handleClose} open={open}>
+      <div className={cn('paper', paperClassName)}>{children}</div>
+    </ModalStyled>
   )
 }
 
 export default GnoModal
```

Comment:
You are leaving this behind just for the purpose to avoid changing all GnoModal for this PR?

Reply:
indeed